src/sdl2/render.rs

```diff
@@ -175,6 +176,29 @@ impl TryFrom<u32> for BlendMode {
     }
 }
 
+#[repr(i32)]
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug)]
+pub enum ScaleMode {
+    /// nearest pixel sampling. default
+    Nearest = SDL_ScaleMode::SDL_ScaleModeNearest as i32,
+    /// linear filtering
+    Linear = SDL_ScaleMode::SDL_ScaleModeLinear as i32,
+    /// anisotropic filtering
+    Best = SDL_ScaleMode::SDL_ScaleModeBest as i32,
+}
+
+impl TryFrom<u32> for ScaleMode {
+    type Error = ();
+
+    fn try_from(n: u32) -> Result<Self, Self::Error> {
+        Ok(match unsafe { transmute(n) } {
```

Comment:
Huge no-no, it will crash badly if `n` is not a safe value, even though the function is a `try_from`, which should simply output a `Err` on invalid value.

Read the implications of transmute with enums and fix this PR, this is not the only time you're using it, and I don't think you should be using it even once.

Reply:
I'm happy to make this change. Quick question first, should these existing instances be changed as well? My intent was to follow the existing style.

- [TextureAccess](https://github.com/Rust-SDL2/rust-sdl2/blob/400e033411b239de10b680f20de800dcb70e9ade/src/sdl2/render.rs#L113)
- [BlendMode](https://github.com/Rust-SDL2/rust-sdl2/blob/400e033411b239de10b680f20de800dcb70e9ade/src/sdl2/render.rs#L167)
- [PixelFormatEnum](https://github.com/Rust-SDL2/rust-sdl2/blob/400e033411b239de10b680f20de800dcb70e9ade/src/sdl2/pixels.rs#L456)
- [EventType](https://github.com/Rust-SDL2/rust-sdl2/blob/400e033411b239de10b680f20de800dcb70e9ade/src/sdl2/event.rs#L334)

Thanks

Reply:
I just checked on the playground, they are all big mistakes and should be fixed ASAP:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1f798ba39aef506228d8ecfeeda9b759

Try in debug and release modes, it's wildly different behaviors, and never what's expected

Reply:
@Cobrand friendly ping!

Reply:
@Cobrand I have addressed cf7d253 one of the job failures.

But the other one is more involved and related to [this](https://github.com/microsoft/vcpkg/issues/43099). Please consider applying the fix listed there. I'd prefer the CI fix to occur in a different MR as it seems out of scope compared to this one (and I can't run the CI manually to check a fix).

Reply:
@Cobrand hello! I noticed that there is some subtle differences between sdl2-sys and sdl3-sys. sdl2-sys causes the casts to be required in this MR. that being said, I'd prefer the scope of this one not to expand too much. Is there anything I can do on my end? I'd love to get this merged in.